optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/drools/uni/DroolsAbstractUniConstraintStream.java

```diff
@@ -38,12 +38,14 @@
 import org.optaplanner.core.api.score.stream.uni.UniConstraintStream;
 import org.optaplanner.core.impl.score.stream.drools.DroolsConstraint;
 import org.optaplanner.core.impl.score.stream.drools.DroolsConstraintFactory;
+import org.optaplanner.core.impl.score.stream.drools.bi.DroolsAbstractBiConstraintStream;
+import org.optaplanner.core.impl.score.stream.drools.bi.DroolsJoinBiConstraintStream;
 import org.optaplanner.core.impl.score.stream.drools.common.DroolsAbstractConstraintStream;
 
 public abstract class DroolsAbstractUniConstraintStream<Solution_, A> extends DroolsAbstractConstraintStream<Solution_>
         implements UniConstraintStream<A> {
 
-    protected final List<DroolsAbstractUniConstraintStream<Solution_, A>> childStreamList = new ArrayList<>(2);
+    protected final List<DroolsAbstractConstraintStream<Solution_>> childStreamList = new ArrayList<>(2);
```

Comment:
Why do we initialize the ArrayList with the exact size of two? Is it guaranteed, that the size will never grow beyond this value?

Reply:
To be honest, I don't know. It's copy-pasted from the Bavet impl.

The size will often be 1. In less frequent cases, 2 or 0. 3, very unlikely. Seems to be a reasonable size.